File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(std::string s) {
        std::unordered_set<char> chSet;
        int l = 0;
        int res = 0;
        for (int r = 0; r < s.length(); ++r) {
            while (chSet.find(s[r]) != chSet.end()) {
                chSet.erase(s[l]);
                l++;
            }
            chSet.insert(s[r]);
            res = std::max(res, r - l + 1);
        }
        return res;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        chSet = set()
        l = 0
        res = 0
        for r in range(len(s)):
            while s[r] in chSet:
                chSet.remove(s[l])
                l+=1
            chSet.add(s[r])
            res = max(res, r-l+1)
        return res